dwds/debug_extension_mv3/tool/build_extension.dart

```
@@ -0,0 +1,56 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// INSTRUCTIONS:
+
+// Builds the unminifed dart2js extension (see DDC issue:
+// see DDC issue: https://github.com/dart-lang/sdk/issues/49869).
+
+// Run from the extension root directory:
+//    - For dev: dart run tool/build_extension.dart
+//    - For prod: dart run tool/build_extension.dart prod
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as p;
+
+const prodFlag = 'prod';
+
+void main(List<String> arguments) {
+  exitCode = 0; // presume success
+  final parser = ArgParser()
+    ..addFlag(prodFlag, negatable: true, defaultsTo: false);
+  final argResults = parser.parse(arguments);
+
+  run(isProd: argResults[prodFlag] as bool);
+}
+
+Future<void> run({required bool isProd}) async {
+  logInfo('Building extension for ${isProd ? 'prod' : 'dev'}');
+  logInfo('Compiling extension with dart2js to /compiled directory');
+  logOutput(
+    await Process.run(
+      'dart',
+      ['run', 'build_runner', 'build', 'web', '--output', 'build', '--release'],
+    ),
+  );
+  logInfo('Updating manifest.json in /compiled directory.');
+  logOutput(
+    await Process.run(
+      'dart',
+      [p.join('tool', 'update_dev_files.dart')],
+    ),
+  );
+}
+
+void logInfo(String message) {
+  print('[BUILD STEP] $message');
+}
+
+void logOutput(ProcessResult result) {
+  final output = result.stdout;
+  final outputList = output is List ? output : [output ?? ''];
+  print(outputList.map((output) => '$output').join('\n'));
```

Can we switch to using transformers on the stream to listen to lines instead and logging them one by one? That way we can see the output in progress, even if the process never ends, for example. Also it will allow us to fail early if needed.

---

Looking at the [documentation](https://api.dart.dev/stable/2.19.2/dart-io/Process-class.html#):
>  "Use the static [start](https://api.dart.dev/stable/2.19.2/dart-io/Process/start.html) and [run](https://api.dart.dev/stable/2.19.2/dart-io/Process/run.html) methods to start a new process. The run method executes the process non-interactively to completion. In contrast, the start method allows your code to interact with the running process."

`Process.start` returns a `Process` with `stdout` and `stderr` streams that can be listened to. I think it makes more sense to use `Process.run` here.